ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/DefaultEnginePipeline.kt

```diff
@@ -61,6 +56,21 @@ public fun defaultEnginePipeline(environment: ApplicationEnvironment): EnginePip
     return pipeline
 }
 
+@InternalAPI
```

Comment:
So these are more like `@EngineAPI` rather than internal, so one who is implementing an engine, should use this function to handler errors, right?

Reply:
Hm, I reused the same visibility as before, but it makes sense to have them visible for other engines. Will update.